cs/client.py

```diff
@@ -432,11 +448,19 @@ def read_config_from_ini(ini_group=None):
         if not conf.has_section(ini_group):
             return dict(name=None)
 
-    all_keys = REQUIRED_CONFIG_KEYS.union(ALLOWED_CONFIG_KEYS)
-    ini_config = {k: v
-                  for k, v in conf.items(ini_group)
-                  if v and k in all_keys}
+    ini_config = {
+        k: v
+        for k, v in conf.items(ini_group)
+        if v and check_key(k, REQUIRED_CONFIG_KEYS.union(ALLOWED_CONFIG_KEYS))
+    }
     ini_config["name"] = ini_group
+
+    # Convert individual header_* settings into a single dict
+    for k in ini_config.copy():
+        if k.startswith("header_"):
+            if "headers" not in ini_config.keys():
+                ini_config["headers"] = {}
```

Comment:
Without the `if` check, you could do:

```python
ini_config.setdefault("headers", {})
```

which would only set the value on first encounter.

Reply:
Thanks, fixed.